modules/evs/fsw/src/cfe_evs_task.c

```diff
@@ -1725,52 +1738,56 @@ int32 CFE_EVS_WriteAppDataFileCmd(const CFE_EVS_WriteAppDataFileCmd_t *data)
-                        break;
+                        /* Clear application file data record */
+                        memset(&AppDataFile, 0, sizeof(CFE_EVS_AppDataFile_t));
+
+                        /* Copy application data to application file data record */
+                        CFE_ES_GetAppName(AppDataFile.AppName, EVS_AppDataGetID(AppDataPtr),
+                                          sizeof(AppDataFile.AppName));
+                        AppDataFile.ActiveFlag           = AppDataPtr->ActiveFlag;
+                        AppDataFile.EventCount           = AppDataPtr->EventCount;
+                        AppDataFile.EventTypesActiveFlag = AppDataPtr->EventTypesActiveFlag;
+
+                        /* Copy application filter data to application file data record */
+                        memcpy(AppDataFile.Filters, AppDataPtr->BinFilters,
+                               CFE_PLATFORM_EVS_MAX_EVENT_FILTERS * sizeof(EVS_BinFilter_t));
+
+                        /* Write application data record to file */
+                        BytesWritten = OS_write(FileHandle, &AppDataFile, sizeof(CFE_EVS_AppDataFile_t));
+
+                        if (BytesWritten == sizeof(CFE_EVS_AppDataFile_t))
+                        {
+                            EntryCount++;
+                        }
+                        else
+                        {
+                            EVS_SendEvent(CFE_EVS_ERR_WRDATFILE_EID, CFE_EVS_EventType_ERROR,
+                                          "Write App Data Command Error: OS_write = 0x%08X, filename = %s",
+                                          (unsigned int)BytesWritten, LocalName);
+                            break;
+                        }
                     }
+                    ++AppDataPtr;
                 }
-                ++AppDataPtr;
-            }
 
-            /* Process command handler success result */
-            if (i == CFE_PLATFORM_ES_MAX_APPLICATIONS)
-            {
-                EVS_SendEvent(CFE_EVS_WRDAT_EID, CFE_EVS_EventType_DEBUG,
-                              "Write App Data Command: %d application data entries written to %s", (int)EntryCount,
-                              LocalName);
-                Result = CFE_SUCCESS;
+                /* Process command handler success result */
+                if (i == CFE_PLATFORM_ES_MAX_APPLICATIONS)
+                {
+                    EVS_SendEvent(CFE_EVS_WRDAT_EID, CFE_EVS_EventType_DEBUG,
+                                  "Write App Data Command: %d application data entries written to %s", (int)EntryCount,
+                                  LocalName);
+                    Result = CFE_SUCCESS;
+                }
             }
-        }
 
-        OS_close(FileHandle);
-    }
+            OS_close(FileHandle);
+        }
 
-    return (Result);
+        return (Result);
 
-} /* End CFE_EVS_WriteAppDataFileCmd */
+    } /* End CFE_EVS_WriteAppDataFileCmd */
```

Comment:
Doesn't look right...

Reply:
Yeah, something went amiss when I rebased across the format change.  I am trying again using another method....